Wikipedia/Code/PlaceSearchService.swift

```diff
@@ -0,0 +1,144 @@
+            let left =  CLLocation(latitude: center.latitude, longitude: center.longitude - halfLongitudeDelta)
+            let right =  CLLocation(latitude: center.latitude, longitude: center.longitude + halfLongitudeDelta)
+            let height = top.distance(from: bottom)
+            let width = right.distance(from: left)
+            
+            let radius = round(0.5*max(width, height))
+            let searchRegion = CLCircularRegion(center: center, radius: radius, identifier: "")
+            
+            locationSearchFetcher.fetchArticles(withSiteURL: siteURL, in: searchRegion, matchingSearchTerm: searchTerm, sortStyle: sortStyle, resultLimit: 50, completion: { (searchResults) in
+                    result = PlaceSearchResult(locationResults: searchResults.results, fetchRequest: nil)
+                    done()
+                }) { (error) in
+
+                    result = PlaceSearchResult(error: error)
+                    done()
+                }
+        }
+    }
+
+    public func fetchSavedArticles(searchString: String?, completion: @escaping (NSFetchRequest<WMFArticle>?) -> () = {_ in }) {
+        let moc = dataStore.viewContext
+        let done = {
+            let request = WMFArticle.fetchRequest()
+            let basePredicate = NSPredicate(format: "savedDate != NULL && signedQuadKey != NULL")
+            request.predicate = basePredicate
+            if let searchString = searchString {
+                let searchPredicate = NSPredicate(format: "(displayTitle CONTAINS[cd] '\(searchString)') OR (snippet CONTAINS[cd] '\(searchString)')")
+                request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [basePredicate, searchPredicate])
+            }
+            request.sortDescriptors = [NSSortDescriptor(key: "savedDate", ascending: false)]
+            
+            completion(request)
+        }
+        
+        do {
+            let savedPagesWithLocation = try moc.fetch(fetchRequestForSavedArticlesWithLocation)
+            guard savedPagesWithLocation.count >= 99 else {
+                let savedPagesWithoutLocationRequest = WMFArticle.fetchRequest()
+                savedPagesWithoutLocationRequest.predicate = NSPredicate(format: "savedDate != NULL && signedQuadKey == NULL")
+                savedPagesWithoutLocationRequest.sortDescriptors = [NSSortDescriptor(key: "savedDate", ascending: false)]
+                let savedPagesWithoutLocation = try moc.fetch(savedPagesWithoutLocationRequest)
+                guard savedPagesWithoutLocation.count > 0 else {
+                    done()
+                    return
+                }
+                let urls = savedPagesWithoutLocation.flatMap({ (article) -> URL? in
+                    return article.url
+                })
+                // TODO: ARM: I don't understand this --------------------V
+                //var allArticlesWithLocation = savedPagesWithLocation // this should be re-fetched
```

Comment:
It meant since we had made a network request and returned to the main queue, those core data objects might have been deleted while we were away, so they weren't safe to use. The comment is invalid now that this code uses `dataStore.viewContext.updateOrCreateArticleSummariesForArticles(withURLs: urls) { (articles)` instead of updating the articles here.